include/supercluster.hpp

```diff
@@ -168,30 +170,26 @@ class Supercluster {
             if (p.zoom <= zoom) continue;
             p.zoom = zoom;
 
-            // find all nearby points
-            std::vector<std::size_t> neighborIds;
-            trees.find(zoom + 1)->second.within(p.x, p.y, r, std::back_inserter(neighborIds));
-
-            bool foundNeighbors = false;
             auto num_points = p.num_points;
             double wx = p.x * num_points;
             double wy = p.y * num_points;
 
-            for (auto id : neighborIds) {
-                auto &b = points[id];
-                // filter out neighbors that are too far or already processed
-                if (zoom < b.zoom) {
-                    foundNeighbors = true;
-                    // save the zoom (so it doesn't get processed twice)
-                    b.zoom = zoom;
-                    // accumulate coordinates for calculating weighted center
-                    wx += b.x * b.num_points;
-                    wy += b.y * b.num_points;
-                    num_points += b.num_points;
-                }
-            }
-
-            if (foundNeighbors) {
+            // find all nearby points
+            trees.find(zoom + 1)->second.within(
+                p.x, p.y, r, [&points, &wx, &wy, &num_points, &zoom](const auto &id) {
```

Comment:
I would write `[&]` here rather than explicitly listing the lambda captures. The lambda will not outlive the call to `within` so there's no danger of dangling captures.